tests/staggered_dslash_ctest.cpp

```diff
@@ -631,20 +274,20 @@ TEST_P(StaggeredDslashTest, benchmark) {
 
     // Clean up loaded gauge field
     for (int dir = 0; dir < 4; dir++) {
-      if (qdp_inlink[dir] != nullptr) { free(qdp_inlink[dir]); qdp_inlink[dir] = nullptr; }
+      if (wrapper.qdp_inlink[dir] != nullptr) { free(wrapper.qdp_inlink[dir]); wrapper.qdp_inlink[dir] = nullptr; }
     }
 
     // Clean up per-partition backup
     for (int p = 0; p < pmax; p++) {
       for (int d = 0; d < 4; d++) {
-        if (qdp_inlink_backup[p][d] != nullptr) { free(qdp_inlink_backup[p][d]); qdp_inlink_backup[p][d] = nullptr; }
-        if (qdp_fatlink_cpu_backup[p][d] != nullptr) {
-          free(qdp_fatlink_cpu_backup[p][d]);
-          qdp_fatlink_cpu_backup[p][d] = nullptr;
+        if (wrapper.qdp_inlink_backup[p][d] != nullptr) { free(wrapper.qdp_inlink_backup[p][d]); wrapper.qdp_inlink_backup[p][d] = nullptr; }
```

Comment:
Likewise can you move these frees into a `ctest_finalize_once`

Reply:
And, similar to the init, it's also important that it gets called exactly once, which is why it can't go into `end()`.